src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,70 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+    private int supplyAmount;
+    private int buyAmount;
+    private int resultAmount;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileContent = readFile(fromFileName);
+        calculateStatistics(fileContent);
+        writeResult(supplyAmount, buyAmount, resultAmount, toFileName);
```

Comment:
You simultaneously use class variables (that will be available in writeResult() method) and pass them as arguments to method.
Generally, when it's not very complicated, it's better to avoid using class variables as when logic growth you will have lots of them and such situation is harder to maintain.

My recommendation is to make calculateStatistics() method to generate ready to write String and pass to writeResult() method only this String and fileName (and then rename calculateStatistics -> generateReport as it's purpose changes).
But if you like solution with class variables it's also ok, then remove redundant arguments from writeResult method header